src/main/java/com/box/sdk/BoxAPIConnection.java

```diff
@@ -22,9 +22,9 @@
  */
 public class BoxAPIConnection {
     /**
-     * The default maximum number of times an API request will be tried when an error occurs.
+     * The default maximum number of times an API request will be retried after an error occurs.
      */
-    public static final int DEFAULT_MAX_ATTEMPTS = 5;
+    public static final int DEFAULT_MAX_RETRIES = 5;
```

Comment:
Setting it to 5 makes the max total attempts 6, which is in line with the box-python-sdk (https://github.com/box/box-python-sdk/blob/9cdebdde39e68e735069dd97e0a459e09382455d/boxsdk/session/session.py#L341)

Reply:
I like the idea of renaming this, but I think changing this value DEFAULT_MAX_RETRIES may be a breaking change. We are changing public values like the the functions to set and get the value. So if we are okay with our next release being a major release, then I am fine with this.

Reply:
Agreed.  I would think the chances someone is using that variable outside the SDK are very low, but if we deprecate it, then we can prevent the breaking change.  See next commit...

Reply:
Makes sense. I think we also need to deprecate `getMaxRequestAttempts` and `setMaxRequestAttempts` in `BoxAPIConnection` and `BoxGlobalSettings`. If we do this, it should be fine. The only other possible issue is the `restore` method in BoxAPIConnection. It takes in an object and restores settings. If someone is storing an object before this change, if they restore with that object there will be an issue. The old object will have the key `maxRequestAttempts`, but our new restore will only look for `maxRetryAttempts`. I don't think this will be a big issue, but not sure.

Reply:
OK, I'll create a separate PR for this refactoring, then.  Thanks!

Reply:
Did the other method deprecations you mentioned.
Also added a test to ensure `restore` works with "old" `state` strings.